lib/model/localizations.dart

```diff
@@ -15,3 +17,60 @@ abstract final class GlobalLocalizations {
   static ZulipLocalizations zulipLocalizations =
     lookupZulipLocalizations(ZulipLocalizations.supportedLocales.first);
 }
+
+/// The mapping, of the locales we offer, to their display name in their own
+/// locale a.k.a. selfname.
+///
+/// [ZulipLocalizations.supportedLocales] can include languages that only have
+/// a few strings translated.  This map should only inlcude sufficiently
+/// translated locales from [ZulipLocalizations.supportedLocales].
+///
+/// The map should be sorted by selfname, to help users find their
+/// language in the UI.  When in doubt how to sort (like between different
+/// scripts, or in scripts you don't know), try to match the order found in
+/// other UIs, like for choosing a language in your phone's system settings.
```

Comment:
Does a map have an order? Maybe we should hard-code a `List<(Locale, String)>`, which is definitely ordered, and compute `kSelfnamesByLocale` from that?

Reply:
And indeed the commit message mentions making a list, but I guess the commit doesn't actually do that 🙂:

```
i18n: Maintain a list of supported languages with metadata
```